### PR TITLE
Star Wars

### PR DESCRIPTION
**Pull request by LeoKuD**
[Link to deploy](https://leokud.github.io/start-wars/)

Done:

- receiving data from API;
- pagination;
- search;
- adaptive;
- save current page after search;
- hide card after click on anothet card;
- loader;
- error processing.

### CODE REVIEW

File: scss/style.scss [hunk 1]
```diff
@@ -1,3 +1,370 @@
...
+  ul {
+    margin: 0;
+    padding: 0;
+    list-style-type: none;
+  }
+}
+
+body {
+  background-color: $bg-color-primary;
+  color: #072227;
```
Comment (maximsan) [inline scss/style.scss:22]: why don't you use variables ? :)

File: scss/style.scss [hunk 2]
```diff
@@ -1,3 +1,370 @@
...
+  background-color: $bg-color-primary;
+  margin: 1rem 0;
+  border-radius: 0.5rem;
+  padding: 0 1rem;
+  text-align: center;
+  cursor: pointer;
+}
+
+.left-menu__item_active {
+  border: 2px solid #072227;
```
Comment (maximsan) [inline scss/style.scss:124]: why don't you use variables ? :)

File: js/library.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const CATEGORIESKEYS = {
+  people: 'people',
+  planets: 'planets',
+  films: 'films',
+  species: 'species',
+  vehicles: 'vehicles',
+  starships: 'starships',
+};
+
+const MENUSTYLING = {
```
Comment (maximsan) [inline js/library.js:57]: rename you constants according to what we discussed

File: js/library.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  leftMenuItemActive: 'left-menu__item_active',
+  leftMenuItem: 'left-menu__item',
+  leftMenuText:'left-menu__text',
+}
+
+const STYLINGPAGINATION = {
```
Comment (maximsan) [inline js/library.js:63]: rename you constants according to what we discussed

File: js/api.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  loader.classList.add(STYLINGLOADER.loaderVisible);
+  fetch(baseURL + path)
+    .then((res) => {
+      return res.json();
+    })
+    .then((data) => {
+    loader.classList.remove(STYLINGLOADER.loaderVisible);
+      cb(data);
+    })
+    .catch((e) => alert('SERVER NOT FOUND'));
```
Comment (maximsan) [inline js/api.js:13]: don't forget to remove you loader if you get an error too

File: js/api.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const baseURL = 'https://swapi.dev/api/';
+
+function getData(path, cb) {
+  loader.classList.add(STYLINGLOADER.loaderVisible);
+  fetch(baseURL + path)
```
Comment (maximsan) [inline js/api.js:5]: let's us string interpolation

File: js/index.js [hunk 7]
```diff
@@ -1,282 +1,184 @@
...
-  e.preventDefault();
-  if (e.submitter.id === btnSubmitForm.id) {
-    taskSubmit();
-  } else if (e.submitter.id === btnSubmitEditForm.id) {
-    submitEditTask(e.submitter.getAttribute('data-index'));
-  }
+  const content = Object.values(CATEGORIESKEYS).map(
+    (item) =>
+      `<li class="left-menu__item" data-category='${item}'><h2 class="left-menu__text">${
+        item[0].toLocaleUpperCase() + item.slice(1)
```
Comment (maximsan) [inline js/index.js:19]: create a method for this and use the method `charAt()` to get element by index in string

File: js/index.js [hunk 8]
```diff
@@ -1,282 +1,184 @@
...
-    tasks.forEach((item, index) => {
-      const taskItem = taskTemplate(item, index);
-      selectedSort === KEYSVARIABLES.up
-        ? !item.isDone
-          ? listCurrentTasks.append(taskItem)
-          : listCompletedTasks.append(taskItem)
-        : !item.isDone
-        ? listCurrentTasks.prepend(taskItem)
-        : listCompletedTasks.prepend(taskItem);
+function fillHtm(data) {
```
Comment (maximsan) [inline js/index.js:32]: `fillHtml` ?

File: js/index.js [hunk 9]
```diff
@@ -1,282 +1,184 @@
...
-function completeTask(index) {
-  tasks[index].isDone = !tasks[index].isDone;
-  updateLocalStorage(LOCALSTORAGEKEYS.tasks, tasks);
-  fillHTMLList();
+  if (paginationArray.length) {
+    const navContent = searchMode
+      ? paginationArray.map(
+          (item) =>
+            `<li data-index='${item}' class='pagination__item ${
+              item === 1 ? 'pagination__item_active' : ''
```
Comment (maximsan) [inline js/index.js:77]: looks like you can use `STYLINGPAGINATION.paginationItemActive`

File: js/index.js [hunk 10]
```diff
@@ -1,282 +1,184 @@
...
+    let items = document.querySelectorAll(STYLINGPAGINATION.paginationItem);
+    items.forEach((elem) =>
+      elem.classList.remove(STYLINGPAGINATION.paginationItemActive)
+    );
+    e.target.classList.add(STYLINGPAGINATION.paginationItemActive);
+    const page = e.target.getAttribute(ATTRIBUTESNAMES.dataIndex);
+    currentPage = parseInt(page);
+    getData(
+      currentCategory +
+        (searchMode ? `/?search=${input.value}&page=` : '/?page=') +
```
Comment (maximsan) [inline js/index.js:102]: create a method to construct search by page part of url based on different params

File: js/index.js [hunk 11]
```diff
@@ -1,282 +1,184 @@
...
-  myForm.elements.priority.forEach((item) => {
-    item.id === tasks[index].priority
-      ? (item.checked = true)
-      : (item.checked = false);
-  });
+function search(e) {
+  e.preventDefault();
+  if (input.value) {
+    searchMode = true;
+    getData(currentCategory + '/' + `?search=${input.value}`, (data) => {
```
Comment (maximsan) [inline js/index.js:115]: Here use the method I advised you to create before

File: js/library.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+const ATTRIBUTESNAMES = {
+  dataIndex: 'data-index',
+  dataIndexasAtr: '[data-index]',
+  itemUrl: 'item-url',
+  dataCategory: 'data-category'
+}
+
+const CARDSTYLING = {
```
Comment (maximsan) [inline js/library.js:78]: use `CLASS` or `CLASS_NAME` part in names like this, because it's more semantically clear

File: index.html [hunk 13]
```diff
@@ -9,7 +9,52 @@
...
+          </nav>
+        </aside>
+        <main class="cards-wrapper">
+        </main>
+      </div>
+      <img src="./img/loader.gif" alt="loader" class="loader">
+    </div>
+
+    <script src="./js/api.js"></script>
+    <script src="./js/library.js"></script>
```
Comment (maximsan) [inline index.html:52]: It's `constants.js`. You keep constants there and name should explain it.

File: scss/style.scss [hunk 14]
```diff
@@ -1,3 +1,370 @@
...
+    font-size: 20px;
+    font-weight: 600;
+
+    text-decoration: none;
+
+    transition-duration: 0.25s;
+  }
+
+  .card {
+    flex-basis: 200px;
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply scss/style.scss]: Fixed. Now only rem.

File: scss/style.scss [hunk 15]
```diff
@@ -1,3 +1,370 @@
...
+  .content-wrapper {
+    flex-direction: column-reverse;
+  }
+
+  .left-menu {
+    position: fixed;
+    top: 26px;
+    left: 20px;
+    width: 26px;
+    height: 26px;
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply scss/style.scss]: Fixed

File: scss/style.scss (hunk 1 above)
Comment (LeoKuD) [reply scss/style.scss:22]:
```suggestion
    color: $font-color-primary;
```

File: scss/style.scss (hunk 2 above)
Comment (LeoKuD) [reply scss/style.scss]: Fixed

File: js/api.js (hunk 6 above)
Comment (LeoKuD) [reply js/api.js:5]:
```suggestion
  fetch(`${baseURL}/${path}`)
```

File: js/api.js (hunk 5 above)
Comment (LeoKuD) [reply js/api.js:13]:
```suggestion
          alert('SERVER NOT FOUND');
      loader.classList.remove(STYLINGLOADER.loaderVisible);
```

File: js/library.js (hunk 3 above)
Comment (LeoKuD) [reply js/library.js:58]:
```suggestion
const LEFT_MENU_CLASS_NAME = {
```
Renamed all constants

File: js/library.js (hunk 4 above)
Comment (LeoKuD) [reply js/library.js:64]:
```suggestion
const PAGINATION_CLASS_NAME = {
```

File: js/index.js (hunk 7 above)
Comment (LeoKuD) [reply js/index.js:18]:
```suggestion
        item.charAt(0).toUpperCase() + item.slice(1)
```

File: js/index.js (hunk 8 above)
Comment (LeoKuD) [reply js/index.js:31]: Yes, but it will better `fillCards`

Comment (LeoKuD) [reply js/index.js:31]:
```suggestion
function fillCards(data) {
```

File: js/index.js [hunk 16]
```diff
@@ -1,282 +1,184 @@
...
-          ? listCurrentTasks.append(taskItem)
-          : listCompletedTasks.append(taskItem)
-        : !item.isDone
-        ? listCurrentTasks.prepend(taskItem)
-        : listCompletedTasks.prepend(taskItem);
+function fillHtm(data) {
+  cardsWrapper.textContent = '';
+  if (data.length) {
+    data.forEach((element, index) => {
+      cardsWrapper.append(createCard(element, index + 1));
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:35]:
```suggestion
      cardsWrapper.append(createCard(element, index++));
```

File: js/index.js [hunk 17]
```diff
@@ -1,282 +1,184 @@
...
-        >
-          Edit
-        </button>`
-        : '<div class="empty-div"></div>'
+function createCard(item, index) {
+  const card = document.createElement('div');
+  card.className = CARDSTYLING.card;
+  const itemUrl = item.url
+    .slice(item.url.lastIndexOf('/', item.url.length - 2))
+    .replace(/\//g, '');
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:47]: This code is no longer needed.

File: js/index.js [hunk 18]
```diff
@@ -1,282 +1,184 @@
...
-          class="btn btn-info w-100 my-2"
-        >
-          Edit
-        </button>`
-        : '<div class="empty-div"></div>'
+function createCard(item, index) {
+  const card = document.createElement('div');
+  card.className = CARDSTYLING.card;
+  const itemUrl = item.url
+    .slice(item.url.lastIndexOf('/', item.url.length - 2))
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:46]: Deleted this code

File: js/index.js [hunk 19]
```diff
@@ -1,282 +1,184 @@
...
+    .replace(/\//g, '');
+  card.setAttribute(ATTRIBUTESNAMES.dataIndex, index);
+  card.setAttribute(ATTRIBUTESNAMES.itemUrl, itemUrl);
+  const cardContent = Object.keys(findKey(currentCategory)).map(
+    (key, index) => {
+      return index === 0
+        ? `<div class='card__title'><h3 class='card__title-text'>${
+            findKey(currentCategory)[key]
+          } : ${item[key]}</h3><div class='card__drop-down'></div></div>`
+        : `<p>${findKey(currentCategory)[key]} : ${item[key]}</p>`;
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:56]:
```suggestion
          const cardContent = selectedCategoryKeysArray.map((key, index) => {
    let name = findKey(currentCategory)[key];
    return index === 0
      ? `<div class='card__title'><h3 class='card__title-text'>${name} : ${item[key]}</h3><div class='card__drop-down'></div></div>`
      : `<p>${name} : ${item[key]}</p>`;
  });
```

File: js/index.js [hunk 20]
```diff
@@ -1,282 +1,184 @@
...
-  tasks.push(new Task([...formValue.values()]));
-  updateLocalStorage(LOCALSTORAGEKEYS.tasks, tasks);
-  fillHTMLList();
-  myForm.reset();
-  btnCloseForm.click();
-}
+function createrPagination(data) {
+  pagination.textContent = '';
+  const paginationArray = [];
+  const countPages = Math.ceil(data.count / 10);
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:68]:
```suggestion
    const countPages = Math.ceil(data.count / numberElementsOnPage);
```

File: js/index.js [hunk 21]
```diff
@@ -1,282 +1,184 @@
...
+          (item) =>
+            `<li data-index='${item}' class='pagination__item ${
+              item === 1 ? 'pagination__item_active' : ''
+            }' >${item}</li>`
+        )
+      : paginationArray.map(
+          (item) =>
+            `<li data-index='${item}' class='pagination__item ${
+              currentPage === item ? 'pagination__item_active' : ''
+            }' >${item}</li>`
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:85]:
```
      const navContent = paginationArray.map(
        (item) =>
          `<li data-index='${item}' class='${
            PAGINATION_CLASS_NAME.paginationItem
          } ${
            (item === currentPage && !searchMode)
              ? PAGINATION_CLASS_NAME.paginationItemActive
              : (item === 1 && searchMode ? PAGINATION_CLASS_NAME.paginationItemActive : '')
          }'>${item}</li>`
      );
```

File: js/index.js (hunk 9 above)
Comment (LeoKuD) [reply js/index.js:78]:
```suggestion
(item === 1 && searchMode ? PAGINATION_CLASS_NAME.paginationItemActive : '')
```

File: js/library.js (hunk 12 above)
Comment (LeoKuD) [reply js/library.js:80]: Fixed. Renemed all constants

File: index.html (hunk 13 above)
Comment (LeoKuD) [reply index.html:52]: Fixed

File: index.html [hunk 22]
```diff
@@ -9,7 +9,52 @@
...
+            <input type="text" class="search__input" placeholder="Search.."/>
+            <i class="fas fa-search"></i>
+          </form>
+        </div>
+      </header>
+
+      <div class="content-wrapper">
+        <aside>
+          <nav class="left-menu">
+            <input id="menu__toggle" type="checkbox" />
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply index.html:34]:
```suggestion
            <input id="menu-toggle" type="checkbox" />
```

File: js/index.js (hunk 10 above)
Comment (LeoKuD) [reply js/index.js]:
```
function setUrl(category, search='', page='') {
  return `${category}${search && `/?search=${search}`}${page && `/?page=${page}`}`
}
```

File: js/index.js (hunk 11 above)
Comment (LeoKuD) [reply js/index.js]:
```
  if (input.value) {
    searchMode = true;
    getData(setUrl(currentCategory, input.value), (data) => {
      fillCards(data.results);
```

File: js/index.js [hunk 23]
```diff
@@ -1 +1,157 @@
...
+      const navContent = paginationArray.map(
+        (item) =>
+          `<li data-index='${item}' class='${
+            PAGINATION_CLASS_NAME.paginationItem
+          } ${
+            item === currentPage && !searchMode
+              ? PAGINATION_CLASS_NAME.paginationItemActive
+              : item === 1 && searchMode
+              ? PAGINATION_CLASS_NAME.paginationItemActive
+              : ''
```
Comment (Dev1LtJ) [inline js/index.js:81]: Do not use nested `?`

File: js/index.js [hunk 24]
```diff
@@ -1 +1,157 @@
...
+    data.forEach((element, index) => {
+      cardsWrapper.append(createCard(element, index++));
+    });
+  } else {
+    cardsWrapper.textContent = 'Nothing found';
+  }
+}
+
+function createCard(item, index) {
+  const selectedCategoryKeysArray = Object.keys(findKey(currentCategory));
```
Comment (Dev1LtJ) [inline js/index.js:44]: create modules (import findkey and other helpers & constants)

File: js/index.js [hunk 25]
```diff
@@ -1 +1,157 @@
...
+
+function togglePages(e) {
+  if (e.target.hasAttribute(ATTRIBUTES_NAME.dataIndex)) {
+    let items = document.querySelectorAll('.pagination__item');
+    items.forEach((elem) =>
+      elem.classList.remove(PAGINATION_CLASS_NAME.paginationItemActive)
+    );
+    e.target.classList.add(PAGINATION_CLASS_NAME.paginationItemActive);
+    const page = e.target.getAttribute(ATTRIBUTES_NAME.dataIndex);
+    !searchMode && (currentPage = parseInt(page));
```
Comment (Dev1LtJ) [inline js/index.js:99]: Try not to use comparisons and assignment in one expression (it's hard to read)

File: js/helpers.js [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+      return SPECIES;
+    case CATEGORIESKEYS.vehicles:
+      return VEHICLES;
+    case CATEGORIESKEYS.starships:
+      return STARSHIPS;
+
+    default:
+      break;
+  }
+}
```
Comment (Dev1LtJ) [inline js/helpers.js:19]: Seems like you can simplify it.

**Review by Dev1LtJ — CHANGES_REQUESTED** [review]
Hello! I left some comments. Please, fix them and let me know.

File: js/index.js (hunk 23 above)
Comment (LeoKuD) [reply js/index.js:81]:
```suggestion
      const navContent = paginationArray.map(
        (item) =>
          `<li data-index='${item}' class='${
            PAGINATION_CLASS_NAME.paginationItem
          } ${
            setActivePaginationItem(item)
          }'>${item}</li>`
      );
```

```
function setActivePaginationItem(item) {
  if (item === 1 && searchMode) {
    return PAGINATION_CLASS_NAME.paginationItemActive
  }
  else if (item === currentPage && !searchMode) {
    return PAGINATION_CLASS_NAME.paginationItemActive
  }
}
```

File: js/index.js (hunk 25 above)
Comment (LeoKuD) [reply js/index.js:99]:
```suggestion
    currentPage = searchMode ? currentPage : parseInt(page);
```

File: js/helpers.js (hunk 26 above)
Comment (LeoKuD) [reply js/helpers.js:19]:
```suggestion
return CATEGORIESKEYS[category]
}
```

File: js/index.js (hunk 24 above)
Comment (LeoKuD) [reply js/index.js:59]: Done

File: js/constants.js [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+export const LEFT_MENU_CLASS_NAME = {
+  leftMenuItemActive: 'left-menu__item_active',
+  leftMenuItem: 'left-menu__item',
+  leftMenuText: 'left-menu__text',
+};
+
+export const PAGINATION_CLASS_NAME = {
+  paginationItem: 'pagination__item',
+  paginationItemActive: 'pagination__item_active',
+  paginationHide: 'pagination_hide',
```
Comment (maximsan) [inline js/constants.js:55]: remove `pagination` part

File: js/constants.js [hunk 28]
```diff
@@ -0,0 +1,72 @@
...
+    manufacturer: 'Manufacturer',
+    cost_in_credits: 'Cost in credits',
+    passengers: 'Passengers',
+  },
+};
+
+export const LEFT_MENU_CLASS_NAME = {
+  leftMenuItemActive: 'left-menu__item_active',
+  leftMenuItem: 'left-menu__item',
+  leftMenuText: 'left-menu__text',
```
Comment (maximsan) [inline js/constants.js:49]: remove `left` part

File: js/constants.js [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+export const PAGINATION_CLASS_NAME = {
+  paginationItem: 'pagination__item',
+  paginationItemActive: 'pagination__item_active',
+  paginationHide: 'pagination_hide',
+};
+export const LOADER_CLASS_NAME = {
+  loaderVisible: 'loader_visible',
```
Comment (maximsan) [inline js/constants.js:58]: remove `loader` part

File: js/constants.js [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+
+export const ATTRIBUTES_NAME = {
+  dataIndex: 'data-index',
+  dataIndexasAtr: '[data-index]',
+  dataCategory: 'data-category',
+};
+
+export const CARD_CLASS_NAME = {
+  card: 'card',
+  cardShowInfo: 'card_show-info',
```
Comment (maximsan) [inline js/constants.js:69]: remove `card` part

File: js/api.js [hunk 31]
```diff
@@ -0,0 +1,20 @@
+import { LOADER_CLASS_NAME } from './constants.js';
+import { cardsWrapper } from './index.js';
+const baseURL = 'https://swapi.dev/api';
+const loader = document.querySelector('.loader');
+
+export function getData(path, cb) {
+  loader.classList.add(LOADER_CLASS_NAME.loaderVisible);
+  fetch(`${baseURL}/${path}`)
+    .then((res) => {
+      return res.json();
```
Comment (maximsan) [inline js/api.js:10]: `() => res.json()`

File: js/api.js [hunk 32]
```diff
@@ -0,0 +1,20 @@
+import { LOADER_CLASS_NAME } from './constants.js';
+import { cardsWrapper } from './index.js';
```
Comment (maximsan) [inline js/api.js:2]: add indentation between  imports and initialisation parts

File: js/index.js [hunk 33]
```diff
@@ -1 +1,171 @@
...
+  card.setAttribute(ATTRIBUTES_NAME.dataIndex, index);
+  const cardContent = selectedCategoryKeysArray.map((key, index) => {
+    let name = selectedCategoryNamesArray[index];
+    return index === 0
+      ? `<div class='card__title'><h3 class='card__title-text'>${name} : ${item[key]}</h3><div class='card__drop-down'></div></div>`
+      : `<p>${name} : ${item[key]}</p>`;
+  });
+  cardContent.push(
+    `<p> <a target='_blank' href='${item.url}'>more info...</> </p>`
+  );
```
Comment (maximsan) [inline js/index.js:72]: method createCardContent(...

File: js/index.js [hunk 34]
```diff
@@ -1 +1,171 @@
...
+      cardsWrapper.append(createCard(element, index++));
+    });
+  } else {
+    cardsWrapper.textContent = 'Nothing found';
+  }
+}
+
+function createCard(item, index) {
+  const selectedCategoryKeysArray = Object.keys(findKey(currentCategory));
+  const selectedCategoryNamesArray = Object.values(findKey(currentCategory));
```
Comment (maximsan) [inline js/index.js:60]: const [keys, names] = Object.entries(findKey(currentCategory))

File: js/index.js [hunk 35]
```diff
@@ -1 +1,171 @@
...
+      paginationArray.push(i);
+    }
+
+    if (paginationArray.length) {
+      const navContent = paginationArray.map(
+        (item) =>
+          `<li data-index='${item}' class='${
+            PAGINATION_CLASS_NAME.paginationItem
+          } ${setActivePaginationItem(item)}'>${item}</li>`
+      );
```
Comment (maximsan) [inline js/index.js:93]: const navContent = pageNumbers.map(createPageItem);

File: js/index.js [hunk 36]
```diff
@@ -1 +1,171 @@
...
+
+function togglePages(e) {
+  if (e.target.hasAttribute(ATTRIBUTES_NAME.dataIndex)) {
+    let items = document.querySelectorAll('.pagination__item');
+    items.forEach((elem) =>
+      elem.classList.remove(PAGINATION_CLASS_NAME.paginationItemActive)
+    );
+    e.target.classList.add(PAGINATION_CLASS_NAME.paginationItemActive);
+    const page = e.target.getAttribute(ATTRIBUTES_NAME.dataIndex);
+    currentPage = searchMode ? currentPage : parseInt(page);
```
Comment (maximsan) [inline js/index.js:109]: currentPage = getCurrentPage();

function getCurrentPage(currentPage = 1) {
    ....
}

**Review by maximsan — CHANGES_REQUESTED** [review]
I left comments. Please fix. Good work!

File: js/constants.js (hunk 27 above)
Comment (LeoKuD) [reply js/constants.js:55]:
```suggestion
  export const PAGINATION_CLASS_NAME = {
  item: 'pagination__item',
  activeItem: 'pagination__item_active',
  hide: 'pagination_hide',
};
```

File: js/constants.js (hunk 28 above)
Comment (LeoKuD) [reply js/constants.js:49]:
```suggestion
export const LEFT_MENU_CLASS_NAME = {
  activeItem: 'left-menu__item_active',
  item: 'left-menu__item',
  text: 'left-menu__text',
};
```

File: js/constants.js (hunk 29 above)
Comment (LeoKuD) [reply js/constants.js:58]:
```suggestion
export const LOADER_CLASS_NAME = {
  visible: 'loader_visible',
};
```

File: js/constants.js (hunk 30 above)
Comment (LeoKuD) [reply js/constants.js:69]:
```suggestion
export const CARD_CLASS_NAME = {
  card: 'card',
  showInfo: 'card_show-info',
  error: 'cards-wrapper_error',
};
```

File: js/api.js (hunk 31 above)
Comment (LeoKuD) [reply js/api.js:10]:
```suggestion
    .then((res) => res.json())
```

File: js/helpers.js [hunk 37]
```diff
@@ -0,0 +1,37 @@
...
+import { searchMode, currentPage, currentCategory, input, fillCards, createrPagination } from "./index.js";
+import { getData } from "./api.js";
+
+export function findKey(category) {
+  return CATEGORIESKEYS[category]
+}
+
+export function createMenuCategory(item) {
+  return `<li class="left-menu__item" data-category='${item}'><h2 class="left-menu__text">${
+    item.charAt(0).toUpperCase() + item.slice(1)
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/helpers.js:11]:
```suggestion
function setCategoryName(item) {
  return `<li class="left-menu__item" data-category='${item}'><h2 class="left-menu__text">${
    item.charAt(0).toUpperCase() + item.slice(1)
  }</h2></li>`;
  
}
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/helpers.js:11]:
```suggestion
  return setCategoryName(item)
```

File: js/api.js (hunk 32 above)
Comment (LeoKuD) [reply js/api.js:2]:
```suggestion
import { cardsWrapper } from './index.js';

```

File: js/index.js (hunk 33 above)
Comment (LeoKuD) [reply js/index.js:72]:
```
export function createCardContent(item) {
  const selectedCategoryKeysArray = Object.keys(findKey(currentCategory));
  const selectedCategoryNamesArray = Object.values(findKey(currentCategory));
  const cardContent = selectedCategoryKeysArray.map((key, index) => {
    let name = selectedCategoryNamesArray[index];
    return index === 0
      ? `<div class='card__title'><h3 class='card__title-text'>${name} : ${item[key]}</h3><div class='card__drop-down'></div></div>`
      : `<p>${name} : ${item[key]}</p>`;
  });
  cardContent.push(
    `<p> <a target='_blank' href='${item.url}'>more info...</> </p>`
  );
  return cardContent.join('')
}
```

File: js/index.js (hunk 34 above)
Comment (LeoKuD) [reply js/index.js:60]: In my case it doesn't work.

File: js/index.js [hunk 38]
```diff
@@ -1 +1,171 @@
...
+  );
+  card.innerHTML = cardContent.join('');
+  return card;
+}
+
+export function createrPagination(data) {
+  if (data.count) {
+    pagination.classList.remove(PAGINATION_CLASS_NAME.paginationHide);
+    pagination.textContent = '';
+    const paginationArray = [];
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:81]: Done.

File: js/index.js (hunk 35 above)
Comment (LeoKuD) [reply js/index.js:93]:
```suggestion
      const navContent = pageNumbers.map(item => createPageItem(item));
```

File: js/index.js [hunk 39]
```diff
@@ -1 +1,171 @@
...
+    });
+  } else {
+    searchMode = false;
+    clearSearch(searchMode);
+  }
+}
+
+function showInfo(e) {
+  const elem = e.target.closest(ATTRIBUTES_NAME.dataIndexasAtr);
+  elem && elem.classList.toggle(CARD_CLASS_NAME.cardShowInfo);
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:132]:
```suggestion
  e.target.closest(ATTRIBUTES_NAME.dataIndexasAtr)?.classList.toggle(CARD_CLASS_NAME.showInfo);
```

File: js/index.js (hunk 36 above)
Comment (LeoKuD) [reply js/index.js:109]:
```suggestion
    currentPage = getCurrntPage(currentPage, searchMode, page);
```

**Comment by LeoKuD** [comment]
Pagination block goes down under cards when adaptiv. Added search button.

File: js/index.js (hunk 34 above)
Comment (maximsan) [reply js/index.js]: why ? from your code I see it should work

**Review by maximsan — APPROVED** [review]
Looks fine for me. Good work! Approved!

File: js/index.js [hunk 40]
```diff
@@ -1 +1,171 @@
...
+    }
+  } else {
+    pagination.classList.add(PAGINATION_CLASS_NAME.paginationHide);
+  }
+}
+
+function togglePages(e) {
+  if (e.target.hasAttribute(ATTRIBUTES_NAME.dataIndex)) {
+    let items = document.querySelectorAll('.pagination__item');
+    items.forEach((elem) =>
```
Comment (Mrio1) [inline js/index.js:97]: Use ```const``` for ```items```.

**Review by Mrio1 — APPROVED** [review]

File: js/index.js (hunk 40 above)
Comment (LeoKuD) [reply js/index.js:97]: Ok, thanks. Fixed.

**Pull request merged**